server/app/controllers/comments_controller.rb

```diff
@@ -9,7 +9,7 @@ def index
     if params[:advert_id].present?
       @comments = Comment.find_by(advert_id: params[:advert_id])
       authorize @comments
-      render json: @comments
+      render json: @comment, serializer: CommentSerializer
```

Comment:
it seems like it should be still @comments, as `set_comment` is not executed for `index` , so that variable @comment is not defined